auslib/db.py

```diff
@@ -1131,33 +1182,28 @@ def _proceedIfNotReadOnly(self, name, limit=None, transaction=None):
 
 class Permissions(AUSTable):
     """allPermissions defines the structure and possible options for all
-       available permissions. Most permissions are identified by an URL,
-       potentially with variables in it. All URL based permissions can be
-       augmented by using the "product" option. When specified, only requests
-       involving the named product will be permitted. Additionally, any URL
-       that supports more than one of: PUT, POST, or DELETE can by augmented
-       by using the option "method". When specified, the permission with this
-       option is only valid for requests through that HTTP method."""
+       available permissions. Permissions can be limited to specific types
+       of actions. Eg: granting the "rule" permission with "actions" set to
+       ["create"] allows rules to be created but not modified or deleted.
+       Permissions that relate to rules or releases can be further limited
+       by product. Eg: granting the "release" permission with "products" set
+       to ["GMP"] allows the user to modify GMP releases, but not Firefox."""
     allPermissions = {
-        'admin': [],
-        '/releases': ['method', 'product'],
-        '/releases/:name': ['method', 'product'],
-        '/releases/:name/revisions': ['product'],
-        '/releases/:name/builds/:platform/:locale': ['method', 'product'],
-        '/releases/:name/read_only': ['method', 'product'],
-        '/rules': ['method', 'product'],
-        '/rules/:id': ['method', 'product'],
-        '/rules/:id/revisions': ['product'],
-        '/users/:id/permissions/:permission': ['method'],
+        "admin": [],
+        "release": ["actions", "products"],
+        "build": ["actions", "products"],
+        "read_only": ["actions", "products"],
```

Comment:
I like those names - I'll update the branch.

You're right that we're only removing unused permissions. We can always add them back later if we want (though one of the advantages of non-URL based permissions is that we can check the "release" permission instead of a history-specific one!).